metrics/metrics.go

```diff
@@ -37,17 +37,36 @@ func init() {
 	for _, enabler := range enablerEnvVars {
 		if val, found := syscall.Getenv(enabler); found && !Enabled {
 			if enable, _ := strconv.ParseBool(val); enable { // ignore error, flag parser will choke on it later
-				log.Info("Enabling metrics collection")
 				Enabled = true
 			}
 		}
 	}
-	for _, arg := range os.Args {
+	for i, arg := range os.Args {
 		flag := strings.TrimLeft(arg, "-")
 
+		// If geth is being configured by toml, try to peak into it.
+		if !Enabled && strings.HasPrefix(flag, "config") && i+1 < len(os.Args) {
```

Comment:
why prefix, why not strict equal?

Reply:
Right, I had the thought that if the style `--config=cfg.toml` is used, might need to handle that differently than just grabbing the next item from args. Was hasty and didn't deal with this before standup, but looking into it more it seemed a bit annoying to deal with all different flag cases so I ended up just rewriting it to use `FlagSet` and I think it now works more reliably.